### PR TITLE
Structured base functions

### PR DESCRIPTION
**Pull request by jnandrawos**
Created Structured Base Functions for all activities and fragments

### CODE REVIEW

File: app/src/main/java/com/example/newsapp/ui/home/fragments/AboutUsFragment.kt [hunk 1]
```diff
@@ -22,11 +22,11 @@ class AboutUsFragment : Fragment() {
     ): View? {
         binding = FragmentAboutUsBinding.inflate(layoutInflater)
 
-        webViewSetup()
+        setupView()
```
Comment (joe-abouassi) [inline app/src/main/java/com/example/newsapp/ui/home/fragments/AboutUsFragment.kt:29]: @jnandrawos can be moved to onViewCreated

File: app/src/main/java/com/example/newsapp/ui/home/fragments/ArticleFragment.kt [hunk 2]
```diff
@@ -22,11 +22,11 @@ class ArticleFragment : Fragment() {
         savedInstanceState: Bundle?,
     ): View? {
         binding = FragmentArticleBinding.inflate(layoutInflater)
-        webViewSetup()
+        setupView()
```
Comment (joe-abouassi) [inline app/src/main/java/com/example/newsapp/ui/home/fragments/ArticleFragment.kt]: @jnandrawos same comment

File: app/src/main/java/com/example/newsapp/ui/home/fragments/ChangePasswordFragment.kt [hunk 3]
```diff
@@ -16,34 +17,44 @@ import dagger.hilt.android.AndroidEntryPoint
...
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?,
     ): View? {
-        val binding: FragmentChangePasswordBinding =
-            FragmentChangePasswordBinding.inflate(layoutInflater, container, false)
+        binding = FragmentChangePasswordBinding.inflate(layoutInflater, container, false)
 
+        implementListeners()
```
Comment (joe-abouassi) [inline app/src/main/java/com/example/newsapp/ui/home/fragments/ChangePasswordFragment.kt:33]: @jnandrawos same

File: app/src/main/java/com/example/newsapp/ui/home/fragments/EditProfileFragment.kt [hunk 4]
```diff
@@ -58,24 +59,32 @@ class EditProfileFragment : Fragment() {
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?,
     ): View? {
-        val binding: FragmentEditProfileBinding =
-            FragmentEditProfileBinding.inflate(layoutInflater, container, false)
+        binding = FragmentEditProfileBinding.inflate(layoutInflater, container, false)
 
+        setupViews()
```
Comment (joe-abouassi) [inline app/src/main/java/com/example/newsapp/ui/home/fragments/EditProfileFragment.kt:62]: @jnandrawos same here

**Review by joe-abouassi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/newsapp/ui/home/fragments/EditProfileFragment.kt (hunk 4 above)
Comment (jnandrawos) [reply app/src/main/java/com/example/newsapp/ui/home/fragments/EditProfileFragment.kt:62]: done

**Review by joe-abouassi — APPROVED** [review]

**Pull request merged**